Modix.Services/Moderation/InfractionSyncingHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Linq;
```

Comment:
I feel like this class just acts as a proxy for `ModerationService.CreateInfractionAsync()`, which does all these checks, essentially making this class redundant.

Would it be better to subscribe to the `UserBanned` notification in `ModerationService` and just reroute to the aforementioned function instead?

Reply:
You could say the same thing about your `StarboardHandler`, in that you could *theoretically* just mash everything together into a single god-service class...

I think the benefit of keeping our handlers in separate classes is in encapsulation and single responsibility. The handler only needs to implement logic specific to its behavior (e.g. determining whether a ban has an associated infraction and creating one if not).

Reply:
I understand what you're saying but this class doesn't accomplish any of that, does it? It merely duplicates the behavior of a class that already exists. It gives the illusion of single responsibility whereas the reality of it being that you're just separating the responsibility of a class that should've had, and can handle this particular behavior as is. What exactly does this class do differently from `ModerationService`s already existing functions?

What I'm trying to get at is that this class doesn't _really_ define any new logic and that's why I deem it unnecessary. All it does is implement a notification-listener which should have been a part of `ModerationService` all along.

Reply:
What part is duplicated?

`ModerationService` doesn't interpret audit logs or generate infractions from Discord bans.

Reply:
I'm a bellend.